src/System.Security.Permissions/src/System/Xaml/Permissions/XamlAccessLevel.cs

```diff
@@ -0,0 +1,150 @@
+            ValidateAssemblyName(assemblyName, "assemblyQualifiedTypeName");
+            
+            return new XamlAccessLevel(assemblyName.FullName, typeName);
+        }
+
+        // Read-only: these properties should not be allowed to be modified once this object
+        // has been passed to XamlLoadPermission
+        
+        // Stored as string: we need to store the assembly and type names, rather than Assembly or
+        // Type references, because permissions can be serialized, and we don't want to force an
+        // assembly load on deserialization in a different AppDomain.
+        
+        public AssemblyName AssemblyAccessToAssemblyName
+        {
+            get { return new AssemblyName(AssemblyNameString); }
+        }
+
+        public string PrivateAccessToTypeName { get; private set; }
+
+        internal string AssemblyNameString { get; private set; }
+
+        internal XamlAccessLevel AssemblyOnly()
+        {
+            return new XamlAccessLevel(AssemblyNameString, null);
+        }
+
+        internal static XamlAccessLevel FromXml(SecurityElement elem)
+        {
+            if (elem.Tag != XmlConstants.XamlAccessLevel)
+            {
+                throw new ArgumentException("", nameof(elem));
+            }
+            
+            string assemblyNameString = elem.Attribute(XmlConstants.AssemblyName);
+            if (assemblyNameString == null)
+            {
+                throw new ArgumentException("", nameof(elem));
+            }
+            AssemblyName assemblyName = new AssemblyName(assemblyNameString);
+            ValidateAssemblyName(assemblyName, "elem");
+
+            string typeName = elem.Attribute(XmlConstants.TypeName);
+            if (typeName != null)
+            {
+                typeName = typeName.Trim();
+            }
+
+            return new XamlAccessLevel(assemblyName.FullName, typeName);
+        }
+
+        internal bool Includes(XamlAccessLevel other)
+        {
+            return other.AssemblyNameString == AssemblyNameString &&
+                (other.PrivateAccessToTypeName == null || other.PrivateAccessToTypeName == PrivateAccessToTypeName);
+        }
+
+        internal SecurityElement ToXml()
+        {
+            SecurityElement element = new SecurityElement(XmlConstants.XamlAccessLevel);
+            element.AddAttribute(XmlConstants.AssemblyName, AssemblyNameString);
+            if (PrivateAccessToTypeName != null)
+            {
+                element.AddAttribute(XmlConstants.TypeName, PrivateAccessToTypeName);
+            }
+            return element;
+        }
+
+        private static void ValidateAssemblyName(AssemblyName assemblyName, string argName)
+        {
+            if (assemblyName.Name == null || assemblyName.Version == null ||
+                assemblyName.CultureInfo == null || assemblyName.GetPublicKeyToken() == null)
+            {
+                throw new ArgumentException("", argName);
```

Comment:
Use `string.Empty`

Reply:
Add strings to https://github.com/dotnet/corefx/blob/master/src/System.Security.Permissions/src/Resources/Strings.resx. Remember to do a local build once to ensure that any XLF tasks are run and localized files are updated

Reply:
Resolving because strings are no longer needed because that functionality was deleted